src/OpenTelemetry.Contrib.Extensions.AWSXRay/Trace/AWSXRayPropagator.cs

```diff
@@ -303,7 +303,13 @@ internal static bool TryParseSampleDecision(ReadOnlySpan<char> sampleDecision, o
     internal static string ToXRayTraceIdFormat(string traceId)
     {
         var sb = new StringBuilder();
+#if NETFRAMEWORK
+#pragma warning disable CA1834
```

Comment:
I think it's better to simply add the version here as a string instead of char even for non-netframework targets. Currently, the version is just one character, it might be updated to something like `1.1` and we would have to update the code here again to make sure that we append the full version. 

Or we could have a check on the version length, if we want to use the optimized version of stringBuilder.Append for a char.

Reply:
The length check doesn't work because the analyser knows it's a constant so still emits a warning.

I've refactored the code to use a `char` for the constant, and then if it needs to change to something like `1.1` then it'll need to change to a `string` with appropriate code changes to keep the analyser happy then.